model/godleyIcon.h

```diff
@@ -78,17 +78,17 @@ namespace minsky
     bool variableDisplay=true;
     void toggleVariableDisplay() {variableDisplay=!variableDisplay;}
     
-    /// scale icon until it's height matches \a h 
-    //void scaleIconForHeight(float h) {update(); m_iconScale*=h/(bottomMargin()+iHeight()*iconScale()*zoomFactor());}        
-    void scaleIconForHeight(float h) {this->scaleFactor(h/(bottomMargin()+iHeight()*this->scaleFactor()*zoomFactor()));}        
+    /// scale icon until it's height matches \a h        
+    void scaleIconForHeight(float h) {update(); this->scaleFactor(Item::scaleFactor()*h/(bottomMargin()+iHeight()*Item::scaleFactor()*zoomFactor()));}        
     
     /// left margin of bank icon with Godley icon
-    float leftMargin() const {return variableDisplay? flowMargin*this->scaleFactor()*zoomFactor(): 0;}
+    float leftMargin() const {return variableDisplay? flowMargin*Item::scaleFactor()*zoomFactor(): 0;}
     /// bottom margin of bank icon with Godley icon
-    float bottomMargin() const {return variableDisplay? stockMargin*this->scaleFactor()*zoomFactor(): 0;}
+    float bottomMargin() const {return variableDisplay? stockMargin*Item::scaleFactor()*zoomFactor(): 0;}
 
     /// icon scale is adjusted when Godley icon is resized
-    //float iconScale() const {return m_iconScale;}
+    float scaleFactor() const override;
+    float scaleFactor(const float& sf) override;
```

Comment:
I'm assuming this override is to prevent updating the boundingBox, is that correct?

Reply:
Yes, because the bounding box was wrong after some resize operations.